### PR TITLE
Feature/7540 speedup q3 verification email banner and modal

### PR DESCRIPTION
**Pull request by DanielPalafox**

### CODE REVIEW

File: src/app/record/components/top-bar-verification-email/modals/top-bar-verification-email-modal/top-bar-verification-email-modal.component.scss [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  margin-inline-end: 10px;
+}
+
+.resend-verification {
+  margin: 20px 0 30px 0;
+}
+
+.need-help {
+  p {
+    margin: 4px 0;
```
Comment (leomendoza123) [inline src/app/record/components/top-bar-verification-email/modals/top-bar-verification-email-modal/top-bar-verification-email-modal.component.scss:17]:
```suggestion
    margin-top: 4px;
    margin-inline-end: 0
```

File: src/app/record/components/top-bar-verification-email/modals/top-bar-verification-email-modal/top-bar-verification-email-modal.component.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  verifyEmail() {
+    this._recordEmailsService
+      .verifyEmail(this.primaryEmail)
+      .pipe(first())
+      .subscribe(() => {
+        this.dialogRef.close()
+      })
+  }
+
```
Comment (leomendoza123) [inline src/app/record/components/top-bar-verification-email/modals/top-bar-verification-email-modal/top-bar-verification-email-modal.component.ts:44]:
```suggestion
  ngOnDestroy() {
    this.$destroy.next(true)
    this.$destroy.unsubscribe()
  }
```

File: src/app/record/components/top-bar-verification-email/modals/top-bar-verification-email-modal/top-bar-verification-email-modal.component.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+@Component({
+  selector: 'app-top-bar-verification-email-modal',
+  templateUrl: './top-bar-verification-email-modal.component.html',
+  styleUrls: [
+    './top-bar-verification-email-modal.component.scss',
+    './top-bar-verification-email-modal.component.scss-theme.scss',
+  ],
+})
+export class TopBarVerificationEmailModalComponent implements OnInit {
```
Comment (leomendoza123) [inline src/app/record/components/top-bar-verification-email/modals/top-bar-verification-email-modal/top-bar-verification-email-modal.component.ts:17]:
```suggestion
export class TopBarVerificationEmailModalComponent implements OnInit, OnDestroy {
```

File: src/app/record/components/top-bar-verification-email/top-bar-verification-email.component.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import { Component, Input, OnInit } from '@angular/core'
```
Comment (leomendoza123) [inline src/app/record/components/top-bar-verification-email/top-bar-verification-email.component.ts:1]:
```suggestion
import { Component, Input, OnInit, OnDestroy } from '@angular/core'
```

File: src/app/record/components/top-bar-verification-email/top-bar-verification-email.component.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+@Component({
+  selector: 'app-top-bar-verification-email',
+  templateUrl: './top-bar-verification-email.component.html',
+  styleUrls: [
+    './top-bar-verification-email.component.scss',
+    './top-bar-verification-email.component.scss-theme.scss',
+  ],
+})
+export class TopBarVerificationEmailComponent implements OnInit {
```
Comment (leomendoza123) [inline src/app/record/components/top-bar-verification-email/top-bar-verification-email.component.ts:17]:
```suggestion
export class TopBarVerificationEmailComponent implements OnInit, OnDestroy {
```

File: src/app/record/components/top-bar-verification-email/top-bar-verification-email.component.ts [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+          this.primaryEmail = primaryEmail.value
+        }
+      })
+  }
+
+  resendVerificationEmailModal() {
+    this._verificationEmailModalService.openVerificationEmailModal(
+      this.primaryEmail
+    )
+  }
```
Comment (leomendoza123) [inline src/app/record/components/top-bar-verification-email/top-bar-verification-email.component.ts:56]:
```suggestion
  }
    ngOnDestroy() {
    this.$destroy.next(true)
    this.$destroy.unsubscribe()
  }
```

File: src/app/record/components/top-bar/top-bar.component.ts [hunk 7]
```diff
@@ -3,13 +3,16 @@ import { UserRecord } from '../../../types/record.local'
...
+import { first, takeUntil } from 'rxjs/operators'
 import { Subject } from 'rxjs'
 import { UserService } from '../../../core'
 import { RecordService } from '../../../core/record/record.service'
 import { isEmpty } from 'lodash'
 import { Assertion, UserInfo } from '../../../types'
 import { UserStatus } from '../../../types/userStatus.endpoint'
+import { RecordEmailsService } from '../../../core/record-emails/record-emails.service'
+import { MatDialog } from '@angular/material/dialog'
+import { VerificationEmailModalService } from '../../../core/verification-email-modal/verification-email-modal.service'
```
Comment (leomendoza123) [inline src/app/record/components/top-bar/top-bar.component.ts:15]:
```suggestion
import { VerificationEmailModalService } from '../../../core/verification-email-modal/verification-email-modal.service'
import { OnDestroy } from '@angular/core'

```

File: src/app/record/components/top-bar/top-bar.component.ts [hunk 8]
```diff
@@ -36,18 +39,32 @@ export class TopBarComponent implements OnInit, OnDestroy {
...
   constructor(
+    private _dialog: MatDialog,
     private _platform: PlatformInfoService,
     private _user: UserService,
-    private _record: RecordService
+    private _record: RecordService,
+    private _recordEmails: RecordEmailsService,
+    private _verificationEmailModalService: VerificationEmailModalService
   ) {
     _platform
```
Comment (leomendoza123) [inline src/app/record/components/top-bar/top-bar.component.ts:56]: We will need to add the onDesroy to this _platform subscription and many others around the app. So I will create a separate card this those.

Comment (leomendoza123) [reply src/app/record/components/top-bar/top-bar.component.ts:56]: https://trello.com/c/UGj8ayuJ/7655-qa-devsmy-orcid-subscriptions-clean-up

Comment (DanielPalafox) [reply src/app/record/components/top-bar/top-bar.component.ts:56]: OK, makes sense

**Comment by DanielPalafox** [comment]
[[speedup Q3] Verification email banner and modal](https://trello.com/c/NTTDNtyc/7540-speedup-q3-verification-email-banner-and-modal)

**Comment by leomendoza123** [comment]
[[QA][my-orcid] Wrong alert shown underneath primary email address when unverified](https://trello.com/c/opGuLiO0/7588-qamy-orcid-wrong-alert-shown-underneath-primary-email-address-when-unverified)

**Pull request merged**